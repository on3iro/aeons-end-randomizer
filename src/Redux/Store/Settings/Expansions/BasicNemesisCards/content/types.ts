import { ActionsUnion } from '@martin_hotell/rex-tils'

import * as types from 'aer-types/types'

import { actions } from './actions'
import { LanguageKey } from '../../types'

type LanguageContent = { [id: string]: types.BasicNemesisCard }

export type State = Readonly<
  {
    [key in LanguageKey]: LanguageContent
  }
>

export enum ActionTypes {}

export type Action = ActionsUnion<typeof actions>

export type BasicNemesisCardsContentStateSlice = {
  Settings: {
    Expansions: {
      BasicNemesisCards: {
        content: State
      }
    }
  }
}
