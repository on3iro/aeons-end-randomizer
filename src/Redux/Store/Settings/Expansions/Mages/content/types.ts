import { ActionsUnion } from '@martin_hotell/rex-tils'

import * as types from 'aer-types/types'

import { actions } from './actions'
import { LanguageKey } from '../../types'

type LanguageContent = { [id: string]: types.Mage }

export type State = Readonly<
  {
    [key in LanguageKey]: LanguageContent
  }
>

export enum ActionTypes {}

export type Action = ActionsUnion<typeof actions>

export type MagesContentStateSlice = {
  Settings: {
    Expansions: {
      Mages: {
        content: State
      }
    }
  }
}
