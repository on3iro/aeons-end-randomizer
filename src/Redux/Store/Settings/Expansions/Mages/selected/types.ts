import { ActionsUnion } from '@martin_hotell/rex-tils'

import { actions } from './actions'

export type State = string[]

export enum ActionTypes {
  TOGGLE = 'Settings/Expansions/Mages/TOGGLE',
  SET_TO_DB = 'Settings/Expansions/Mages/SET_TO_DB',
  SET_TO_DB_SUCCESS = 'Settings/Expansions/Mages/SET_TO_DB_SUCCESS',
  SET_TO_DB_FAILURE = 'Settings/Expansions/Mages/SET_TO_DB_FAILURE',
  FETCH_FROM_DB = 'Settings/Expansions/Mages/FETCH_FROM_DB',
  FETCH_FROM_DB_SUCCESS = 'Settings/Expansions/Mages/FETCH_FROM_DB_SUCCESS',
  FETCH_FROM_DB_FAILURE = 'Settings/Expansions/Mages/FETCH_FROM_DB_FAILURE',
}

export type Action = ActionsUnion<typeof actions>

export type SelectedMagesStateSlice = {
  Settings: {
    Expansions: {
      Mages: {
        selected: string[]
      }
    }
  }
}
