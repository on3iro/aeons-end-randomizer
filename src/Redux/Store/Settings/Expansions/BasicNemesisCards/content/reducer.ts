import { LoopReducer } from 'redux-loop'

import AERData from 'aer-data/src/index'

import { Action, State } from './types'

export const initialState: State = {
  ENG: AERData.normalizedData.ENG.basicNemesisCards,
  FR: AERData.normalizedData.FR.basicNemesisCards,
  PL: AERData.normalizedData.PL.basicNemesisCards,
  DE: AERData.normalizedData.DE.basicNemesisCards,
}

export const Reducer: LoopReducer<State, Action> = (
  state: State = initialState,
  action: Action
) => {
  switch (action.type) {
    default: {
      return state
    }
  }
}
