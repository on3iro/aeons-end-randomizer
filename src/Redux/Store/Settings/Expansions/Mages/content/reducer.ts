import { LoopReducer } from 'redux-loop'

import AERData from 'aer-data/src/index'

import { Action, State } from './types'

export const initialState: State = {
  ENG: AERData.normalizedData.ENG.mages,
  FR: AERData.normalizedData.FR.mages,
  PL: AERData.normalizedData.PL.mages,
  DE: AERData.normalizedData.DE.mages,
}

export const Reducer: LoopReducer<State, Action> = (
  state: State = initialState,
  action: Action
) => {
  switch (action.type) {
    default: {
      return state
    }
  }
}
