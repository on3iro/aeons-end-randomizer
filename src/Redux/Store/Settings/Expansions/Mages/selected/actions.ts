import { createAction } from '@martin_hotell/rex-tils'

import { ActionTypes, State } from './types'

export const actions = {
  noOp: () => createAction('@@REDUX_LOOP/ENFORCE_DEFAULT_HANDLING'),
  toggle: (id: string) => createAction(ActionTypes.TOGGLE, id),
  setToDB: (state: State) => createAction(ActionTypes.SET_TO_DB, state),
  setToDBSuccessful: () => createAction(ActionTypes.SET_TO_DB_SUCCESS),
  setToDBFailed: (error: Object) =>
    createAction(ActionTypes.SET_TO_DB_FAILURE, error),
  fetchFromDB: () => createAction(ActionTypes.FETCH_FROM_DB),
  fetchFromDBSuccessful: (selectedMages: unknown) =>
    createAction(ActionTypes.FETCH_FROM_DB_SUCCESS, selectedMages),
  fetchFromDBFailed: (error: Object) =>
    createAction(ActionTypes.FETCH_FROM_DB_FAILURE, error),
}
