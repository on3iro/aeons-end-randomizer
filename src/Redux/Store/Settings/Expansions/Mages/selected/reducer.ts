import { LoopReducer } from 'redux-loop'

import AERData from 'aer-data/src/index'

import { Action, ActionTypes, State } from './types'

import * as reducerHelpers from './reducerHelpers'

export const initialState: State = AERData.normalizedData.ENG.mageIds

export const Reducer: LoopReducer<State, Action> = (
  state: State = initialState,
  action: Action
) => {
  switch (action.type) {
    case ActionTypes.TOGGLE: {
      return reducerHelpers.toggle(state, action)
    }

    case ActionTypes.FETCH_FROM_DB: {
      return reducerHelpers.fetchFromDb(state)
    }

    case ActionTypes.FETCH_FROM_DB_SUCCESS: {
      return reducerHelpers.fetchFromDbSuccess(state, action)
    }

    case ActionTypes.FETCH_FROM_DB_FAILURE: {
      return state
    }

    default: {
      return state
    }
  }
}
