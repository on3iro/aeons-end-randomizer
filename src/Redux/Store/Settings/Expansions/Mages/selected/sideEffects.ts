import { get as getFromDb, set as setToDb } from 'idb-keyval'
import { MAGES_DB_KEY } from '../constants'
import { State } from './types'

export const getSelectedMagesFromDB = () => getFromDb(MAGES_DB_KEY)

export const setSelectedMagesToDB = (selectedMages: State) =>
  setToDb(MAGES_DB_KEY, selectedMages)
