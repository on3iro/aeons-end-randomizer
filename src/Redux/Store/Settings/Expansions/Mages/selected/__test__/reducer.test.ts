import { Cmd, getCmd, getModel } from 'redux-loop'

import { State } from '../types'
import { actions } from '../actions'
import { getSelectedMagesFromDB, setSelectedMagesToDB } from '../sideEffects'

import { initialState, Reducer } from '../reducer'

const mockSelectedMagesState: State = initialState

describe('Settings | Expansions | Mages | selected | reducer', () => {
  it('should return the initial state', () => {
    // @ts-ignore
    const result = Reducer(undefined, {})

    expect(result).toEqual(initialState)
  })

  it('should handle TOGGLE', () => {
    const selectedExpansionsToSave = initialState.filter(
      (mage) => mage !== 'Adelheim'
    )

    const result = Reducer(mockSelectedMagesState, actions.toggle('Adelheim'))

    const model = getModel(result)
    const cmd = getCmd(result)

    expect(model).toMatchSnapshot()
    expect(cmd).toEqual(
      Cmd.run(setSelectedMagesToDB, {
        args: [selectedExpansionsToSave],
        successActionCreator: actions.setToDBSuccessful,
        failActionCreator: actions.setToDBFailed,
      })
    )
  })

  it('should handle FETCH_FROM_DB', () => {
    const result = Reducer(initialState, actions.fetchFromDB())

    const model = getModel(result)
    const cmd = getCmd(result)

    expect(model).toEqual(initialState)

    expect(cmd).toEqual(
      Cmd.run(getSelectedMagesFromDB, {
        successActionCreator: actions.fetchFromDBSuccessful,
        failActionCreator: actions.fetchFromDBFailed,
      })
    )
  })

  it('should handle FETCH_FROM_DB_SUCCESS for defined state', () => {
    const result = Reducer(
      mockSelectedMagesState,
      actions.fetchFromDBSuccessful(['Adelheim'])
    )

    expect(getModel(result)).toEqual(['Adelheim'])
  })

  it('should handle FETCH_FROM_DB_SUCCESS for undefined state', () => {
    const result = Reducer(
      initialState,
      // @ts-ignore
      actions.fetchFromDBSuccessful(undefined)
    )

    expect(getModel(result)).toEqual(initialState)
  })

  it.todo('should handle FETCH_FROM_DB_FAILURE')
  it.todo('should handle SET_TO_DB')
  it.todo('should handle SET_TO_DB_SUCCESS')
  it.todo('should handle SET_TO_DB_FAILURE')
})
