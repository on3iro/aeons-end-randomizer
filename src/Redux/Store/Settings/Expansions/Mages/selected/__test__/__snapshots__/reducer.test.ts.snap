// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings | Expansions | Mages | selected | reducer should handle TOGGLE 1`] = `
Array [
  "Brama",
  "Jian",
  "Kadir",
  "Lash",
  "Mist",
  "Phaedraxa",
  "Xaxos",
  "Nym",
  "Reeve",
  "Zhana",
  "Dorian",
  "Inco",
  "Razra",
  "LegacyGreen",
  "LegacyYellow",
  "LegacyOrange",
  "LegacyBlue",
  "Sahala",
  "Gygar",
  "Soskel",
  "Taqren",
  "Talix",
  "Rhia",
  "Claudia",
  "Lost",
  "Malastar",
  "Indira",
  "Remnant",
  "Auren",
  "Willow",
  "Ona",
  "Nadea",
  "Kavoc",
  "Bayli",
  "Shoshana",
  "Rhys",
  "Naftir",
  "Lilly",
  "Nook",
  "Mazra",
  "Qu",
  "Sparrow",
  "XaxosTV",
  "Dezmodia",
  "Garu",
  "Gex",
  "Mazahaedron",
  "MistWE",
  "Quilius",
  "Ulgimor",
  "YanMagda",
  "Ilya",
  "Kel",
  "ZhanaO",
  "TaqrenO",
  "QuO",
  "Thraxir",
  "DezmodiaO",
  "Arachnos",
  "KiriAndPhoenix",
  "OhatAndUlgimor",
  "Cairna",
  "Lucien",
  "Reth",
  "ArachnosLOG",
  "ClaudiaLOG",
  "DezmodiaLOG",
  "GygarLOG",
  "IlyaLOG",
  "IncoLOG",
  "IndiraLOG",
  "KadirLOG",
  "KelLOG",
  "LostLOG",
  "MalastarLOG",
  "MazraLOG",
  "NookLOG",
  "QuLOG",
  "RazraLOG",
  "SoskelLOG",
  "TalixLOG",
  "TaqrenLOG",
  "ThraxirLOG",
  "XaxosLOG",
  "YanMagdaLOG",
]
`;
